code-challenges.js

```diff
@@ -9,67 +9,186 @@
 // Use the test variables provided below. Expected output: "true is not a string" "42 is not a string" "mCdngMstr"
 
+//Pseudocode
+//refactor function to look for different data types
+//add an if statement to test if the type of data is a string
+//add an else statement that runs the code from the previous question and outputs just the vowels.
+
 var refactorTester1 = true
 var refactorTester2 = 42
 var refactorTester3 = "IAmACodingMaster"
 
+const refactorTest = (data) => {
+    if (typeof data !== "string") {
+        return `${data} is not a string`
+    } else {
+    let stringToArray = data.split("")
+    return stringToArray.filter(value => {
+        return value !== 'a' && value !== 'e' && value !== 'i' && value !== 'o' && value !== 'u'
+    }).join("")
+    }
+}
 
+console.log(refactorTest(refactorTester1))
+console.log(refactorTest(refactorTester2))
+console.log(refactorTest(refactorTester3))
 
 
 
 // --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
 // Expected output: "chopsticks" "stemware"
 
+//Pseudocode
+//create a function named kitchenItems that takes in an array of strings as an argument
+//declare a variable itemsSorted and apply the .sort() method to return itemsSorted in ascending order
+    //after googling a bunch of different things the .sort() method was suggested by someone on discord
+    //had to learn how to use it to sort the items in the array by length
+//use .pop() method to remove and return the last value in the sorted array
+
+
 var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
 var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
 
+const longestString = (array) => {
+    let itemsSorted = array.sort(function(a, b) {
+        return a.length - b.length;
+    })
+    return itemsSorted.pop()
+}
 
-
+console.log(longestString(utensils))
+console.log(longestString(vessels))
 
 
 // --------------------7) Create a function that takes in an array of strings and returns an array with the last letter removed from each string.
 // Expected output: ["Luig", "Peac", "Toa", "Yosh"] ["Ink", "Blink", "Pink", "Clyd"]
 
+//Pseudocode
+//create a function that takes in an argument of an array
+//return a new array using .map() 
+//inside .map() use .substr() method to return a string starting at the 0 index -1 of the length to remove the last letter.
+    //had to learn about the .substr() method that returns a portion of the string starting at the specified index and extending for a given number of charachters afterwards. Using the .length method like in the last problem to access the length of the string and then subtract 1 to remove the last lettter.
+
 var mario = ["Luigi", "Peach", "Toad", "Yoshi"]
 var pacman = ["Inky", "Blinky", "Pinky", "Clyde"]
+
+const lessLastLetter = (array) => {
```

Comment:
keep up the descriptive function naming. General convention is to start with a verb that describes what the function is doing